### ISSUE TITLE
restic hides and destroys directories with the same name (data is lost)

### ISSUE THREAD

**Issue by luizluca** [restic/restic#1179]
I might be doing something just stupid but I guess this is what users generally do.

If I backup two different paths that have the same final directory name, restic simply backups the first one and ignores all the other ones. Or better, it do save them, but all commands that could read data ignore them and prune gets rid of them.

## Output of `restic version`
```
restic 0.7.1
compiled with go1.8.3 on linux/amd64
```

## How did you run restic exactly?

```
echo 123 > restic.key
mkdir repo
mkdir -p home/test1/data
date > home/test1/data/testfile1.txt
mkdir -p home/test2/data
date > home/test2/data/testfile2.txt
mkdir -p home/test3/data
date > home/test3/data/testfile3.txt
restic -p restic.key -r repo init
...

restic -p restic.key -r repo backup home/test1/data home/test2/data home/test3/data
scan [/root/a/home/test1/data /root/a/home/test2/data /root/a/home/test3/data]
scanned 3 directories, 3 files in 0:00
[0:00] 100.00%  0B/s  90B / 90B  6 / 6 items  0 errors  ETA 0:00 
duration: 0:00, 0.00MiB/s
snapshot 8bfeec2b saved
```
Until here, everything was OK. It saved 6 items (3 directories and 3 files).
Wait, 3 directories? Wasn't it supposed to be 6 directories?

```
restic -p restic.key -r repo snapshots
ID        Date                 Host        Tags        Directory
----------------------------------------------------------------------
8bfeec2b  2017-08-26 00:52:00  zipper              ┌── /root/a/home/test1/data
                                                   │   /root/a/home/test2/data
                                                   └── /root/a/home/test3/data
```
Still OK.

```
restic -p restic.key -r repo ls latest
snapshot 8bfeec2b of [/root/a/home/test1/data /root/a/home/test2/data /root/a/home/test3/data] at 2017-08-26 00:52:00.186855416 -0300 -03):
/data
/data/testfile1.txt
```

I guess that wasn't what I really asked it to backup. Did it really saved only data from test1?

```
restic -p restic.key -r repo find testfile1.txt
Found matching entries in snapshot 8bfeec2b4574226f411df235a6ef51d26da8ee635dd670d862950bce8206a371
```

testfile1.txt is there, but testfile2.txt ...

```
restic -p restic.key -r repo find testfile2.txt
<nothing returned!>
```

At least, mount, ls, find and specially, restore, never listed testefile2.txt or testfile3.txt.
Let's debug:

```
restic -p restic.key -r repo list blobs
tree ab1808987f20ebf517b2f2b906d73b74f0be79d701786979f2f525f7281d902e
data 711db6309812a19a5535d1b4c2302ced3e2f4f1d656b39ff0c088d4eabfe7b1a
data 7ba9ce1f2c12857a31b629c2c1f9826a0b45147900024534159cb908a2576737
tree c67214e81e8380335629696f2056575dc324ae2980f97098a2219c29e268a57e
tree 6b257a61f0197ff8ca9a82b1358dcedf08b9ec6d1744d28348ef992b75da2bf8
tree d147d18cfeeac8146d00e1f98925b29e179193b54561983a77c244b53dbd0d4b
```

Four dirs and 2 files? Maybe two of them are equal?

```
md5sum home/test*/*/*
7dd0c1b72dc0f35688241f7ac4e42504  home/test1/data/testfile1.txt
7dd0c1b72dc0f35688241f7ac4e42504  home/test2/data/testfile2.txt
42b2d1966ff6645925190b3db15b32e9  home/test3/data/testfile3.txt
```

Yes, it did deduplicated it nicely. Let's show all of them:

```
for blob in $(restic -p restic.key -r repo list blobs | awk '{ print $2 }'); do restic -p restic.key -r repo cat blob $blob; done
{"nodes":[{"name":"data","type":"dir","mode":2147484141,"mtime":"2017-08-26T00:51:50.489170628-03:00","atime":"2017-08-26T00:52:00.185075187-03:00","ctime":"2017-08-26T00:51:50.489170628-03:00","uid":0,"gid":0,"inode":12486783,"device_id":23,"content":null,"subtree":"d147d18cfeeac8146d00e1f98925b29e179193b54561983a77c244b53dbd0d4b"}]}
sáb ago 26 00:51:50 -03 2017
sáb ago 26 00:51:49 -03 2017
{"nodes":[{"name":"testfile2.txt","type":"file","mode":420,"mtime":"2017-08-26T00:51:49.713178185-03:00","atime":"2017-08-26T00:52:00.185075187-03:00","ctime":"2017-08-26T00:51:49.713178185-03:00","uid":0,"gid":0,"inode":12486781,"device_id":23,"size":30,"links":1,"content":["7ba9ce1f2c12857a31b629c2c1f9826a0b45147900024534159cb908a2576737"]}]}
{"nodes":[{"name":"testfile1.txt","type":"file","mode":420,"mtime":"2017-08-26T00:51:49.709178223-03:00","atime":"2017-08-26T00:52:00.189075147-03:00","ctime":"2017-08-26T00:51:49.709178223-03:00","uid":0,"gid":0,"inode":12486778,"device_id":23,"size":30,"links":1,"content":["7ba9ce1f2c12857a31b629c2c1f9826a0b45147900024534159cb908a2576737"]}]}
{"nodes":[{"name":"testfile3.txt","type":"file","mode":420,"mtime":"2017-08-26T00:51:50.49317059-03:00","atime":"2017-08-26T00:52:00.189075147-03:00","ctime":"2017-08-26T00:51:50.49317059-03:00","uid":0,"gid":0,"inode":12486784,"device_id":23,"size":30,"links":1,"content":["711db6309812a19a5535d1b4c2302ced3e2f4f1d656b39ff0c088d4eabfe7b1a"]}]}
```

Everything is correctly saved, except that there is only one 'data' directory. And finally, let's try prune:

```
restic -p restic.key -r repo prune
counting files in repo
building new index for repo
[0:00] 100.00%  1 / 1 packs
repository contains 1 packs (6 blobs) with 1.842 KiB bytes
processed 6 blobs: 0 duplicate blobs, 0B duplicate
load all snapshots
find data that is still in use for 1 snapshots
[0:00] 100.00%  1 / 1 snapshots
found 3 of 6 data blobs still in use, removing 3 blobs
will remove 0 invalid files
will delete 0 packs and rewrite 1 packs, this frees 818B
[0:00] 100.00%  1 / 1 packs rewritten
counting files in repo
[0:00] 100.00%  1 / 1 packs
finding old index files
saved new index as be5f649a
remove 1 old index files
[0:00] 100.00%  1 / 1 packs deleted
done
```

There goes all my files that "restic ls" didn't list...

I guess this is **really** not expected.

**Comment by luizluca** [comment]
The problem goes one... if I repeat the backup, it skips the first data directory as it did not change but add the second data as if data directory was changed.

**Comment by dionorgua** [comment]
probably #549

**Comment by luizluca** [comment]
@dionorgua, yes, that is the issue.

It's a too serious bug to have already more than one year. This leads to data loss.

**Comment by fd0** [comment]
You're right, thanks for reporting this again. This is a serious issue and in between #549 and now it became worse: Before, restic created a tree with several entries with the same name, so the data is saved and still referenced, but couldn't easily be restored.

I'm going to close it as a duplicate of #549 and see what we can do about it near-term. Feel free to add further comments, thanks!

**Comment by fd0** [comment]
This is resolved in #1209.
